lib/schema_registry/client.rb

```diff
@@ -55,8 +66,54 @@ def default_compatibility_level=(level)
       request(:put, "/config", compatibility: level)
     end
 
+    # Build options hash for net/http based on params provided. Primary for selectivly adding TLS config options for MTLS
+    def build_ssl_options
```

Comment:
What is the value add of this method? What if we would ask for an argument, that we simply forward to `Net::HTTP` as is, rather than constructing our own? `Net::HTTP` can be a bit awkward, but it's very well documented so it should be relatively easy for people to figure things out.

Reply:
We could always add a helper function if we consider "raw" `Net::HTTP` to cumbersome.

``` ruby
# raw
Client.new('https://foo.bar', use_ssl: true, verify_mode: OpenSSL::SSL::VERIFY_PEER)

# with helper
Client.new('https://foo.bar', Client.connection_options(...))
```

Reply:
Something like this:

1. Make the client class as straightforward as possible, simply forwarding stuff to Net::HTTP
2. Create a helper method that makes it easier to instantiate such a class.

E.g. the class would look something like this:

``` ruby
class Client
  def initialize(endpoint, username = nil, password = nil, connection_options = {}
    ...
    @connection_options = connection_options
    ...
  end

  def request
    ...
    Net::HTTP.start(endpoint.host, endpoint.port, @connection_options) do |http| ... end
  end
end
```

For the helper method, you could either make a static method that returns a hash that you could use for the `connection_options` argument. Or you could create additional constructors.

``` ruby
class Client
  def self.connection_options(...)
    ...
    return { ... }
  end
end

Client.new('https://foo.bar', 'user', 'pass', Client.connection_options(...))
```

or, as an additional constructor:

``` ruby
class Client
  def self.from_config(configuration)
    ...
    new(endpoint, username password, connection_options_from_config)
  end
end
```

I'm fine with either option, so whatever you think is best. As long as the user is able to simply provide options that will be passed to `Net::HTTP`, and can set any option as they see fit, our API will never be limiting them.